src/request.rs

```diff
@@ -277,6 +277,47 @@ impl RequestBuilder {
         Ok(self)
     }
 
+    /// Sends a multipart/formdata body.
+    ///
+    /// ```no_run
+    /// use reqwest::mime;
+    /// use reqwest::{MultipartRequest, MultipartField, to_multipart};
+    ///
+    /// let client = reqwest::Client::new().unwrap();
+    /// let response = client.post("your url").unwrap()
+    ///     .multipart(
+    ///         // Add fields from anything serializable
+    ///         to_multipart(&[("key", "value"), ("key2", "value2")]).unwrap()
+    ///         // Add fields builder style
+    ///         .field(MultipartField::param("key3", "value3"))
+    ///         .field(MultipartField::param("json", "{ \"number\": 5 }")
+    ///             .mime(Some(mime::APPLICATION_JSON)))
+    ///         .field(MultipartField::file("file", "/path/to/file")
+    ///             .expect("File could not be opened"))
+    ///     ).send();
+    ///
+    /// ```
+    ///
+    /// See [`to_multipart`](fn.to_multipart.html), [`MultipartRequest`](struct.MultipartRequest.html)
+    /// and [`MultipartField`](struct.MultipartField.html) for more examples.
+    pub fn multipart(&mut self, multipart: MultipartRequest) -> &mut RequestBuilder {
```

Comment:
I dont know Serde well enough to know if that is possible. I feel it does not work because the thing implements Serialize could be a list, struct or hashmap. How would MultipartField be added to a struct or hashmap?